src/journey-maps/angular/journey-maps.ts

```diff
@@ -559,8 +563,13 @@ export class SbbJourneyMaps implements OnInit, AfterViewInit, OnDestroy, OnChang
       });
     }
 
-    if (changes.styleOptions?.currentValue.mode !== changes.styleOptions?.previousValue?.mode) {
-      this._mapStyleModeChanged.next();
+    if (
```

Comment:
Would it be easier to read with

    JSON.stringify(changes.styleOptions?.currentValue) !== JSON.stringify(changes.styleOptions?.previousValue)